# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [23]:
import pandas as pd
import numpy as np
import math
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [12]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

<b>1.What test is appropriate for this problem? Does CLT apply?</b>

The two sample test (white sample versus black sample) that compares the proportions of candidates who got the call is appropriate since the two samples are independent of each other and among themselves.

The sample size is,

In [11]:
len(data[data.race=='b'])

2435

In [7]:
len(data[data.race=='w'])

2435

Hence we can apply CLT since the sample size is comfortably greater than 1000

<b>2.What are the null and alternate hypotheses?</b>

Null hypothesis: Race has no significant impact on whether or not a candidate gets called. More specifically, the true population proportion of black folks who got a call and the true population proportion of white folks who got a call is the same (difference is zero)

Alternate hypothesis: Race has a significant impact on whether or not a candidate gets called. More specifically, there is a difference in the true population proportions.

<b>3. Compute margin of error, confidence interval, and p-value</b>

Proportion of Black applicants who got a call back in the given sample is,

In [14]:
Pblack = sum(data[data.race=='b'].call)/len(data[data.race=='b'])
print Pblack

0.064476386037


And the corresponding sampling distribution variance as per CLT is (since n is large we can assume the sample mean to be the popuation mean)

In [17]:
Vblack = Pblack*(1 - Pblack)/(len(data[data.race=='b']))
print Vblack

2.47717378565e-05


Proportion of White applicants who got a call back in the given sample is,

In [15]:
Pwhite = sum(data[data.race=='w'].call)/len(data[data.race=='w'])
print Pwhite

0.0965092402464


And the corresponding sampling distribution variance as per CLT is

In [18]:
Vwhite = Pwhite*(1 - Pwhite)/(len(data[data.race=='w']))
print Vwhite

3.5809119833e-05


The variance of the distribution of the difference in proportions of the White sample and the Black sample is

In [20]:
Vtotal = Vblack + Vwhite
print Vtotal

6.05808576895e-05


Hence the standard deviation of the sampling distribution of the difference in proportions (standard error) is

In [24]:
SDtotal = math.sqrt(Vtotal)
print SDtotal

0.00778337058668


In [26]:
d = 1.96 * SDtotal
print d

0.0152554063499


Margin of error for 95% level of confidence is: plus or minus 0.01526

In [28]:
Psample = Pwhite - Pblack
print Psample

0.0320328542094


In [29]:
Psample + d

0.047288260559332024

In [30]:
Psample - d

0.016777447859559147

95% confidence interval is 0.0168 to 0.0473

Now assuming that the null hypothesis is true and there is no difference in the proportions and Pwhite = Pblack then each of the proportion is

In [37]:
Pnull = (sum(data[data.race=='b'].call) + sum(data[data.race=='w'].call))/len(data)
print Pnull

0.0804928131417


The standard error is

In [38]:
SDnull = math.sqrt(2 * Pnull * (1 - Pnull)/len(data[data.race=='w']))
print SDnull

0.00779689403617


In [39]:
z = (Psample - 0)/SDnull
print z

4.10841215243


In [50]:
1 - stats.norm(0,1).cdf(z)

1.9919434187887219e-05

The p-value is 1.9919434187887219e-05 which is very less than the 0.05 significance level

<b>4. Write a story describing the statistical significance in the context or the original problem.</b>

There is a 95% chance that if a random sample is chosen from the labor population of the United States then the ratio of the folks with White sounding names in the sample getting a call will be greater than the folks with Black sounding names. We can state with a reasonable degree of confidence (95% to be precise) and a 100% sense of disappointment that a candidate with a White sounding name is 1.68% to 4.73% more likely to be called as compared to a candidate with a Black sounding name.

<b>5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?</b>

No. Although race does seem to matter we need to take a more wholistic approach and consider race in conjunction with the other factors such as years of experience, education etc. It could be that Black individuals may be having less years of experience overall or less years of education compared to their White counterparts which may be the reason for getting lesser number of calls. Hence infering a racial bias only by considering one variable ('race') would be incorrect. The analysis could be ammended by taking into account other variables available in the data.